### PR TITLE
GetOptions/DeleteOptions

### PR DESCRIPTION
**Pull request by ncdc**

### CODE REVIEW

File: pkg/client/deploymentconfigs.go [hunk 1]
```diff
@@ -61,7 +62,7 @@ func (c *deploymentConfigs) List(opts metainternal.ListOptions) (result *deploya
 }
 
 // Get returns information about a particular deploymentConfig
-func (c *deploymentConfigs) Get(name string) (result *deployapi.DeploymentConfig, err error) {
+func (c *deploymentConfigs) Get(name string, options *metav1.GetOptions) (result *deployapi.DeploymentConfig, err error) {
```
Comment (sttts) [inline pkg/client/deploymentconfigs.go]: `VersionedParams(options, kapi.ParameterCodec).` is missing

File: pkg/client/egressnetworkpolicy.go [hunk 2]
```diff
@@ -50,7 +51,7 @@ func (c *egressNetworkPolicy) List(opts metainternal.ListOptions) (result *sdnap
 }
 
 // Get returns information about a particular firewall
-func (c *egressNetworkPolicy) Get(name string) (result *sdnapi.EgressNetworkPolicy, err error) {
+func (c *egressNetworkPolicy) Get(name string, options *metav1.GetOptions) (result *sdnapi.EgressNetworkPolicy, err error) {
 	result = &sdnapi.EgressNetworkPolicy{}
 	err = c.r.Get().Namespace(c.ns).Resource("egressNetworkPolicies").Name(name).Do().Into(result)
```
Comment (sttts) [inline pkg/client/egressnetworkpolicy.go]: here too

File: pkg/client/appliedclusterresourcequota.go [hunk 3]
```diff
@@ -40,8 +41,8 @@ func (c *appliedClusterResourceQuotas) List(opts metainternal.ListOptions) (resu
 }
 
 // Get returns information about a particular appliedClusterResourceQuota and error if one occurs.
-func (c *appliedClusterResourceQuotas) Get(name string) (result *quotaapi.AppliedClusterResourceQuota, err error) {
+func (c *appliedClusterResourceQuotas) Get(name string, options metav1.GetOptions) (result *quotaapi.AppliedClusterResourceQuota, err error) {
 	result = &quotaapi.AppliedClusterResourceQuota{}
-	err = c.r.Get().Namespace(c.ns).Resource("appliedclusterresourcequotas").Name(name).Do().Into(result)
+	err = c.r.Get().Resource("appliedclusterresourcequotas").Name(name).VersionedParams(&options, kapi.ParameterCodec).Do().Into(result)
```
Comment (sttts) [inline pkg/client/appliedclusterresourcequota.go]: the dropped namespace is intentional?

Comment (ncdc) [reply pkg/client/appliedclusterresourcequota.go]: No

File: pkg/authorization/registry/clusterpolicy/registry.go [hunk 4]
```diff
@@ -130,3 +130,15 @@ func (s ReadOnlyClusterPolicy) List(options metainternal.ListOptions) (*authoriz
 func (s ReadOnlyClusterPolicy) Get(name string, options *metav1.GetOptions) (*authorizationapi.ClusterPolicy, error) {
 	return s.GetClusterPolicy(apirequest.WithNamespace(apirequest.NewContext(), ""), name, options)
 }
+
+type ReadOnlyClusterPolicyClientShim struct {
```
Comment (sttts) [inline pkg/authorization/registry/clusterpolicy/registry.go:134]: can we shorten the name to ClientShim?

Comment (ncdc) [reply pkg/authorization/registry/clusterpolicy/registry.go:134]: Ok

File: pkg/client/hostsubnets.go [hunk 5]
```diff
@@ -47,7 +48,7 @@ func (c *hostSubnet) List(opts metainternal.ListOptions) (result *sdnapi.HostSub
 }
 
 // Get returns host subnet information for a given host or an error
-func (c *hostSubnet) Get(hostName string) (result *sdnapi.HostSubnet, err error) {
+func (c *hostSubnet) Get(hostName string, options metav1.GetOptions) (result *sdnapi.HostSubnet, err error) {
```
Comment (sttts) [inline pkg/client/hostsubnets.go:51]: Is one missing here?

Comment (ncdc) [reply pkg/client/hostsubnets.go:51]: One what?

Comment (ncdc) [reply pkg/client/hostsubnets.go:51]: Oh, VersionedParams, yes. My sed expected "name string" and hostName messed it up.

Comment (ncdc) [reply pkg/client/hostsubnets.go:51]: Fixed

**Pull request merged**